public/app/features/plugins/extensions/usePluginComponents.tsx

```diff
@@ -38,13 +37,16 @@ export function usePluginComponents<Props extends object = {}>({
       extensionPointId,
     });
 
-    // Only log error for an invalid `extensionPointId` in DEV mode
-    if (enableRestrictions && !isExtensionPointIdValid({ extensionPointId, pluginId })) {
-      pointLog.error(errors.INVALID_EXTENSION_POINT_ID);
+    // Don't show extensions if the extension-point id is invalid in DEV mode
+    if (isGrafanaDevMode() && !isExtensionPointIdValid({ extensionPointId, pluginId, isInsidePlugin, log: pointLog })) {
+      return {
+        isLoading: false,
+        components: [],
+      };
```

Comment:
Bare with me because I'm not that familiar with these hooks but it seems like we have good test coverage for the validators which is awesome. I'm wondering if these `if statements` within the hooks are quite similar between all these hooks and if they could be refactored out to something that is easier to test? Seems like we're repeating the code both in hooks and tests now.

Reply:
Very good point @hugohaggmark! Yeah, I agree, there's a lot of repetition in the hooks that could be refactored and thus make the logical flow easier to maintain and understand. I will create an issue for it and it to our board. 👍

Reply:
Created an issue, please feel free to update it Hugo if I have missed something:
https://github.com/grafana/grafana/issues/107965

Reply:
Awesome thanks 👍